### PR TITLE
Ticket26 form navigation and exit modal

### PR DESCRIPTION
**Pull request by yndajas**
# Summary

Merge #7 (ticket 10) first, as this will probably have conflicts to resolve

## Functionality

The buttons that interact with the survey will currently just `console.log` (the actual methods we'll need to use are provided as comments for now until this is connected up with #7), but the buttons that open modals work

## Other additions/edits

- added `.nvmrc` - not sure if everyone is using 14.18.1 but that version is working for me, whereas later versions aren't
- updated `Gemfile.lock` and `yarn.lock` for compatibility

## To do?

- I added modified icons with different colours - ideally these would be components that just use `currentColor`, but I couldn't work out how to do that in this Rails Vue setup. Any ideas, or shall we leave this for now?
- I also noticed the designs are using Helvetica Neue, but we'd probably need to change that site-wide if we use it?

# Testing
- load server
- navigate to [/criteria/new](localhost:3000/criteria/new)
  -  check navigation against [design](https://xd.adobe.com/view/85f38f01-ca5a-4c6b-a393-28b15a7e278b-d0a0/specs)
  - click 'Exit' button to launch modal and check against [design](https://xd.adobe.com/view/85f38f01-ca5a-4c6b-a393-28b15a7e278b-d0a0/screen/d5e07126-6e55-4030-b926-e60ebd8e3351/specs)
- navigate to [/commitments/new](localhost:3000/commitments/new)
  -  check navigation against [design](https://xd.adobe.com/view/85f38f01-ca5a-4c6b-a393-28b15a7e278b-d0a0/screen/9c7965c3-64f7-46a7-bd7d-8188712d59eb/specs)
  - click 'Exit' button to launch modal and check against [design](https://xd.adobe.com/view/85f38f01-ca5a-4c6b-a393-28b15a7e278b-d0a0/screen/0cac2fea-f68a-41ec-83bf-4c117ff9052e/specs)
- in `/app/javascript/components/form/SurveyForm.vue`, change `isFirstPage` to `false` (keeping `isLastPage` set to `false`) to check against middle page commitment form designs (e.g. [location page](https://xd.adobe.com/view/85f38f01-ca5a-4c6b-a393-28b15a7e278b-d0a0/screen/9de804d6-0cb2-4d31-b51d-80ef1d67d15e/specs))
- in `/app/javascript/components/form/SurveyForm.vue`, change `isFirstPage` to `false` and `isLastPage` to `true` to check against [last page commitment form design](https://xd.adobe.com/view/85f38f01-ca5a-4c6b-a393-28b15a7e278b-d0a0/screen/69cf8c9f-4a32-4f5e-939c-e4474644615f/specs)

### CODE REVIEW

File: app/javascript/packs/application.js [hunk 1]
```diff
@@ -37,6 +38,7 @@ document.addEventListener('DOMContentLoaded', () => {
       store,
       components: {
 	      FilteredTable,
```
Comment (sergiomarrocoli) [inline app/javascript/packs/application.js:40]: weird indent here, but only on github. looks fine in vscode  ¯\_(ツ)_/¯

**Comment by sergiomarrocoli** [comment]
looks good to me. only thing in the UI that's a bit off is the font (but maybe this is something we're doing later/it's close enough), and the modal backdrop looks lighter than in the mock ups

File: app/javascript/packs/application.js (hunk 1 above)
Comment (yndajas) [reply app/javascript/packs/application.js:40]: Yeah I noticed that!

**Review by sergiomarrocoli — APPROVED** [review]

File: app/assets/stylesheets/components/form/_navigation.scss [hunk 2]
```diff
@@ -0,0 +1,73 @@
+//--------------------------------------------------
+// form navigation
+//--------------------------------------------------
+
+.form-navigation__items {
```
Comment (RainbowGiantSquid) [inline app/assets/stylesheets/components/form/_navigation.scss:5]: whitespace (as if extra line above class name)

File: app/javascript/components/forms/Modal.vue [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+          />
+          <button
+            class="modal__button modal__button--continue"
+            v-text="modalText.continue"
+            @click="closeModal"
+          />
+          <button
+            class="modal__button modal__button--exit"
+            v-text="modalText.exit"
+            @click="exit"
```
Comment (RainbowGiantSquid) [inline app/javascript/components/forms/Modal.vue:24]: could we make this button functional? So maybe add a link to locales and pass it here.

File: app/javascript/components/forms/SurveyForm.vue [hunk 4]
```diff
@@ -10,30 +24,56 @@ import Turbolinks from "turbolinks";
 import axios from "axios";
 import { setAxiosHeaders } from "../../helpers/axios-helpers";
 import "survey-vue/modern.css";
```
Comment (RainbowGiantSquid) [inline app/javascript/components/forms/SurveyForm.vue:26]: I don't have a strong preference but if you prefer using single quotes in js part I think let's change it all to single quotes.

**Review by RainbowGiantSquid — CHANGES_REQUESTED** [review]
All works great, just a thing to add and tiny formatting thing. Also second Sergio on the font, so far I can see everything is set up in Arial, I've posted in the channel to ask if we have a licence to use Helvetica Neue so we can leave it for now I think.

File: app/javascript/packs/application.js (hunk 1 above)
Comment (RainbowGiantSquid) [reply app/javascript/packs/application.js]: Happened to me, copy and paste the whole components block sorted it out.

File: app/javascript/components/forms/Modal.vue (hunk 3 above)
Comment (yndajas) [reply app/javascript/components/forms/Modal.vue:24]: Didn't add to locales as it's either going to call `survey.completeLastPage()` or redirect to `/dashboard` (depending on the form), but I've added that functionality. It currently works on the criteria form, but we need to config hash setting up for it to work on the commitments form

File: app/javascript/components/forms/SurveyForm.vue (hunk 4 above)
Comment (yndajas) [reply app/javascript/components/forms/SurveyForm.vue:26]: Updated the components edited in this branch to use double quotes consistently

**Review by RainbowGiantSquid — APPROVED** [review]

**Pull request merged**